File: server/util/env.go
package util

import "os"

// ProjectID returns the GCP Project ID.
func ProjectID() string {
	return os.Getenv("GOOGLE_CLOUD_PROJECT")
}

// ClientOrigin returns the origin for client.
func ClientOrigin() string {
	origin := os.Getenv("CLIENT_ORIGIN")
	if origin != "" {
		return origin
	}
	return "https://" + ProjectID() + ".appspot.com"
}

// APIOrigin returns the origin for api.
func APIOrigin() string {
	origin := os.Getenv("API_ORIGIN")
	if origin != "" {
		return origin
	}
	return "https://api-dot-" + ProjectID() + ".appspot.com"
}

File: server/web/env.rb
# frozen_string_literal: true

module Web
  module Env
    PROJECT_ID = ENV['GOOGLE_CLOUD_PROJECT']
    API_ORIGIN = ENV['API_ORIGIN'] || 'localhost:8080'
    CLIENT_ORIGIN = ENV['CLIENT_ORIGIN'] || 'localhost:3000'
    API_VERSION = 'v1'
  end
end
